### ISSUE TITLE
--help showing more than just the first comment block

### ISSUE THREAD

**Issue by wolfmah** [ianthehenry/sd#10]
Hi there!
While using `sd` I found a bug with the `--help` command when trying to display the first comment block from a script.

My scripts are not all one-liner (or close too). Some are pretty long-winded, and they end up with a bunch a comments to help the future me remember what is going on. I don't want to keep the help in a sidecar file (though it's nice for the folders), so I put all the juicy info in the first comment block. But the `sd z --help` is outputting everything that is a comment starting at a line, not just the first block.

**Script file**
<details>
    <summary><code>~/sd/z</code></summary>

```shell
#!/usr/bin/env bash

# First line
# Second line
#
# Final line

set -eu

#####
# The following variables are doing stuff
AAA="aaa"
BBB="bbb"

echo "$AAA $BBB"
```
</details>

**Result with original `sd`**
<details>
    <summary><code>sd z --help</code></summary>

```
First line
Second line

Final line
####
###
##
#

The following variables are doing stuff
```
</details>

Kinda jarring! So I dug around a little and found that the code doing the parsing of the comments is this: https://github.com/ianthehenry/sd/blob/dd645e48bf951a5a944e102de451e469030023b3/sd#L102

In my local copy, I tweaked some stuff around and ended up with the following replacement for line 102:
```shell
help=$(awk -v shebangFound=false -v commentFound=false '/^\s*#.*$/ { if ($0 ~ /^\s*#\!/) { shebangFound = "true"; } else { commentFound = "true"; print $0; }; }; !/^#.*$/ { if (shebangFound == "true" && commentFound == "true") { exit; }; }' "$target" | sed -E 's/^# ?//')
```

**Result after my tweak**
<details>
    <summary><code>sd z --help</code></summary>

```
First line
Second line

Final line
```
</details>

It holds up anything that I could throw at it that I already had in my bag of shell/python scripts. Though I haven't really tried to put it through ringer by crafting some really nasty input.

Knowing that this project is pretty far on the "personal software" (as opposed to "general purpose"), this is more of a FYI to help others if they bump into that sort of bug. If you decide to implement it, or tweak it, all the better! 👍

**Comment by ianthehenry** [comment]
Hey thanks! Sorry I didn't have time to look at this before now. This is very much a dumb bug that I've fixed in v1.1.0 (using more horrible unreadable sed instead of your nice awk script, for some reason). Hopefully it has the exact same effect, but if you upgrade please let me know if you spot any other  issues with it.

**Comment by wolfmah** [comment]
The new release works like a charm! Thanks you for taking the time to look into it.
